core/shared/src/test/scala/laika/parse/code/languages/LanguageSpec.scala

```diff
@@ -642,6 +642,55 @@ class LanguageSpec extends AnyWordSpec with Matchers {
         keyword("PRIMARY"), space, keyword("KEY"), other(" ("), id("`BAZ_ID`"), other(")\n);")
       )
     }
+
+    "parse a YAML document" in {
+      val input =
+        """# Doc
+          |
+          |```yaml
+          |# Source: mychart/templates/configmaps.yaml
+          |apiVersion: v1
+          |kind: 'ConfigMap'
+          |metadata:
+          |  name: port_overriden # Some comment
+          |  tag: abc#def
+          |data:
+          |  host: "localhost"
+          |  port: 9876 # That's fine
+          |  timeout: 23.56
+          |  null_value: null
+          |  boolean: true
+          |a_sequence:
+          |  - Item 1
+          |  - 123
+          |  - key: value
+          |```
+        """.stripMargin
+
+      val nl = other("\n")
+      val nl2 = other("\n  ")
+      val colon = other(": ")
+      parse(input) shouldBe result("yaml",
+        comment("# Source: mychart/templates/configmaps.yaml\n"),
+        attrName("apiVersion"), colon, string("v1"), nl,
+        attrName("kind"), colon, string("'ConfigMap'"), nl,
+        attrName("metadata"), other(":\n  "),
+        // It would be more accurate if the space of `other(" ")` was part of its preceding string("port_overriden")
+        // but in the end it does not matter how space will be colored
```

Comment:
I'd suggest to remove this comment as it is misleading. In fact, the way you implemented it is exactly correct for two reasons: a) comparable highlighters in Laika behave in the same way, b) a YAML parser would not include the whitespace before the `#` in the result.